### ISSUE TITLE
Build ML + Deep Learning Model

### ISSUE THREAD

**Issue by ryantusi** [ryantusi/Ship_Delay_AI_Predictor#2]
- Implement Feature Engineering and Process Data to create model
- Split data into Train, Test, and Validation Sets
- Build ML models with Logistic Regression, Decision Trees, Random Forest, Gradient Boosting, SVM, and K-Nearest Neighbor
- Evaluate and compare ML models 
- Tune Hyperparameters
- Build Deep Learning Models with Tensorflow -> SimpleNN, DeepNN, and ResNet
- Evaluate and compare DL Models
- Implement Ensemble Methods (ML + DL Fusion)
- Select Final Best Performance Model
- Save Models in in h5 and joblib. Store metadata in json.
- Test with sample predictions
- Dimensionality Reduction using PCA
